moped-editor/src/views/dashboard/DashboardView.js

```diff
@@ -125,47 +211,66 @@ const DashboardView = () => {
+  // sets the data grid row data when query data is fetched
+  useEffect(() => {
+    let selectedData = [];
+
+    if (TABS[activeTab].label === "Following" && !!data) {
+      selectedData = data.moped_user_followed_projects;
+    } else if (TABS[activeTab].label === "My projects" && !!data) {
+      selectedData = data.moped_proj_personnel;
+    }
+
+    if (selectedData) {
       /**
-       * Get percentage of milestones completed
+       * Build data needed in Dashboard Material Table
        */
-      const milestonesTotal = project.project.moped_proj_milestones.length;
-      const milestonesCompleted = project.project.moped_proj_milestones.filter(
-        (milestone) => milestone.completed === true
-      ).length;
-      project["completed_milestones_percentage"] = !!milestonesTotal
-        ? (milestonesCompleted / milestonesTotal) * 100
-        : 0;
+      selectedData.forEach((project) => {
+        project["project_name_full"] = project.project.project_name_full;
+        project["project_id"] = project.project.project_id;
+        project["phase_name"] =
+          project.project.moped_proj_phases?.[0]?.moped_phase.phase_name;
+        project["phase_key"] =
+          project.project.moped_proj_phases?.[0]?.moped_phase.phase_key;
+        project["current_phase_id"] =
+          project.project.moped_proj_phases?.[0]?.moped_phase.phase_id;
+        /**
+         * Get percentage of milestones completed
+         */
+        const milestonesTotal = project.project.moped_proj_milestones.length;
+        const milestonesCompleted =
+          project.project.moped_proj_milestones.filter(
+            (milestone) => milestone.completed === true
+          ).length;
+        project["completed_milestones_percentage"] = !!milestonesTotal
+          ? (milestonesCompleted / milestonesTotal) * 100
+          : 0;
 
-      // project status update equivalent to most recent project note
-      // html is parsed before being rendered in the DashboardStatusModal component
-      project["status_update"] = "";
-      if (project?.project?.moped_proj_notes?.length) {
-        const note = project.project.moped_proj_notes[0]["project_note"];
-        project["status_update"] = note ? note : "";
-      }
-    });
-  }
+        project["id"] = project.project.project_id;
+
+        // project status update equivalent to most recent project note
+        // html is parsed before being rendered in the DashboardStatusModal component
+        project["status_update"] = "";
+        if (project?.project?.moped_proj_notes?.length) {
+          const note = project.project.moped_proj_notes[0]["project_note"];
+          project["status_update"] = note ? note : "";
+        }
+      });
+    }
+    if (data) {
+      setRows(selectedData);
+    }
+  }, [data, activeTab]);
+
+  const handleRowModesModelChange = (newRowModesModel) => {
+    setRowModesModel(newRowModesModel);
+  };
 
   /** Build custom user greeting
    */
```

Comment:
What do y'all think about pulling this custom user greeting code outside of the component definition? its currently being recreated on every render I believe

Reply:
i think this is a great idea!